testcases/OpTestSTB.py

```diff
@@ -0,0 +1,623 @@
+        if int(l_out) == 0:
+            print "Trusted Boot device-tree file exists"
+            return BMC_CONST.FW_SUCCESS
+        else:
+            print "No Trusted Boot device-tree file exists"
+            return BMC_CONST.FW_FAILED
+
+    ##
+    # @brief This function checks for the device-tree node of Secure boot
+    # Returns FW_SUCCESS when SB_DT_ENTRY found
+    # Returns FW_FAILED when SB_DT_ENTRY not found
+    #
+    def proc_sb_verify(self):
+        l_cmd = "test -f %s; echo $?" % BMC_CONST.SB_DT_ENTRY
+        l_out = self.cv_HOST.host_run_command(l_cmd)
+        l_out = l_out.strip("\n\r")
+        if int(l_out) == 0:
+            print "Secure Boot device-tree file exists"
+            return BMC_CONST.FW_SUCCESS
+        else:
+            print "No Secure Boot device-tree file exists"
+            return BMC_CONST.FW_FAILED
+
+    ##
+    # @brief This function cross verifies the output from hw_sb_support function
+    # which reads Secure boot value from SSR value and secure_boot_mode function
+    # which reads Secure boot value from OPAL messagelog
+    # When both the values are not same then raises OpTestError
+    #
+    def cross_check_hw_opal_sb(self):
+        if self.hw_sb_support() == self.secure_boot_mode():
+            print "For Secure boot Both SSR value and OPAL are in sync"
+        else:
+            l_msg = "For Secure boot Both SSR value and OPAL are NOT in sync"
+            print l_msg
+            raise OpTestError(l_msg)
+
+    ##
+    # @brief This function cross verifies the output from hw_tb_support function
+    # which reads Trusted boot value from SSR value and trusted_boot_mode
+    # function, which reads Trusted boot value from OPAL messagelog
+    # When both the values are not same then raises OpTestError
+    #
+    def cross_check_hw_opal_tb(self):
+        if self.hw_tb_support() == self.trusted_boot_mode():
+            print "For Trusted boot Both hw and opal are in sync"
+        else:
+            l_msg = "For Trusted boot Both hw and opal are not in sync"
+            print l_msg
+            raise OpTestError(l_msg)
+
+    ##
+    # @brief This function cross verifies the output from different functions,
+    # namely capp_measured, bootkernel_measured,
+    # capp_verified, bootkernel_verified,
+    # capp_loaded, bootkernel_loaded,
+    # And the above values are cross checked with the Secure and Trusted boot
+    # modes. As given below.
+    #
+    # When Secureboot is ON, Trustedboot is ON
+    # Partitions are Verified, Measured, Loaded
+    #
+    # When Secureboot if OFF, Trustedboot is ON
+    # Partitions are NOT Verified, Measured, Loaded
+    #
+    # When Secureboot is ON, Trustedboot is OFF
+    # Partitions are Verified, NOT Measured, Loaded
+    #
+    # When secureboot is OFF, trustedboot is OFF
+    # Partitions are NOT Verified, NOT Measured, Loaded
+    #
+    # In the case of mismatches raises OpTestError
+    #
+    def cross_check_stb(self):
```

Comment:
you can keep the below code as it is, but change it into two testcases opal_verify_capp_part & opal_verify_bootkernel_part, so that it is easy to track failures for individual resources.